dll/win32/browseui/basebarsite.cpp

```diff
@@ -381,9 +381,9 @@ HRESULT STDMETHODCALLTYPE CBaseBarSite::SetDeskBarSite(IUnknown *punkSite)
     if (punkSite == NULL)
     {
 
-        TRACE("Destroying site \n");
+        TRACE("Destroying site\n");
         /* Cleanup our bands */
-        while(SUCCEEDED(EnumBands(-1, &dwBandID)) && dwBandID)
+        while (EnumBands(-1, NULL))
         {
             hResult = EnumBands(0, &dwBandID);
             if(FAILED_UNEXPECTEDLY(hResult))
```

Comment:
```suggestion
            if (FAILED_UNEXPECTEDLY(hResult))
```

Reply:
Is it possible we could enter an infinite loop if the EnumBands(-1) returns a non zero count, but a EnumBands(0) call fails?

Reply:
Possible, yes, probably.

Reply:
Would it be possible to do instead:
```c
while ((hResult = EnumBands(0, &dwBandID)))
{
...
}
```
without the part that queries the number of (remaining) bands, without loosing generality?

Reply:
You are missing a SUCCEEDED but yes, you could do that. But I just changed it to a loop that counts down instead, this will try to remove all bands even if some fail.